bundles.go

```diff
@@ -126,16 +127,64 @@ func NewBundle() (bundle *Bundle, err error) {
 	return
 }
 
+// Reference ukify from systemd:
+// https://github.com/systemd/systemd/blob/d09df6b94e0c4924ea7064c79ab0441f5aff469b/src/ukify/ukify.py
+
 func GenerateBundle(bundle *Bundle) (bool, error) {
-	args := []string{
-		"--add-section", fmt.Sprintf(".osrel=%s", bundle.OSRelease), "--change-section-vma", ".osrel=0x20000",
-		"--add-section", fmt.Sprintf(".cmdline=%s", bundle.Cmdline), "--change-section-vma", ".cmdline=0x30000",
-		"--add-section", fmt.Sprintf(".linux=%s", bundle.KernelImage), "--change-section-vma", ".linux=0x2000000",
-		"--add-section", fmt.Sprintf(".initrd=%s", bundle.Initramfs), "--change-section-vma", ".initrd=0x3000000",
+	type section struct {
+		section string
+		file    string
+	}
+	sections := []section{
+		{".osrel", bundle.OSRelease},
+		{".cmdline", bundle.Cmdline},
+		{".splash", bundle.Splash},
+		{".linux", bundle.KernelImage},
```

Comment:
The kernel needs a buffer area after it, or there is a chance it clobbers the initrd when run through efi handover protocol. In systemd we add it last to hopefully avoid this case. (This is only for old kernels.)

Reply:
How do we fix that? How much buffer space does it need?

Also, wouldn't it make more sense to keep `.linux` after `.initrd`?